docs/user-guide/src/capm3/node_reuse.md

```diff
@@ -0,0 +1,45 @@
+# Node Reuse
+
+This feature brings a possibility of re-using the same BaremetalHosts (referred to as a host later)
+during deprovisioning/provisioning mainly as part of the rolling upgrade process in the cluster.
+
+## Workflow
+
+Metal3MachineTemplate (M3MT) CRD is the object to support enabling/disabling node reuse feature. When set to **true**,
+Cluster API Provider Metal3 (CAPM3) Machine controller performs the following:
+
+During deprovisioning:
+
+1. Finds M3MT referenced by Metal3Machine (M3M) and checks if nodeReuse is enabled;
+1. If enabled, it checks Cluster API (CAPI) machine object role (control-plane or worker), and based on that, fetches
+   either KubeadmControlPlane (KCP) or MachineDeployment (MD) name accordingly;
+1. Sets fetched object's name on the corresponding host's labels field.
```

Comment:
Would be better to explain the label format here: <infrastructure.cluster.x-k8s.io/node-reuse: kcp-object-name>

Reply:
I would keep it as is since it is already detailed and more than that can be found in the original proposal of the feature.